sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -232,10 +225,53 @@ def _ReadEndpoint(self):
         return self._global_endpoint_manager.get_read_endpoint()
 
     async def _setup(self):
```

Comment:
is this a new function added? when it will called?

Reply:
yeah, this function is an addition to the async client that is necessary in order to properly warm up the client - it gets called when a client gets initialized using `async with` or when a user calls `client.__aenter__()` to manually warm it up

Reply:
gotcha, thanks for the explanation